### PR TITLE
Modified single page op

### PR DESCRIPTION
**Pull request by abuts**
The `apply_c` code modified for simplicity, some changes from prototype review are applied, design complete.

### CODE REVIEW

File: horace_core/utilities/compute_bin_data.m [hunk 1]
```diff
@@ -16,8 +16,8 @@
 % variance  The variance array to calculate binning
 %
 % no_normalization -- if present and true, do not normalize signal and
-%            variance by npix to calculate proper averages. Return sums
-%            only
+%            variance by npix to calculate proper averages. Returns bin sums
+%            only. If provided, variance have to be provided (may be empty, not to use it)
```
Comment (cmarooney-stfc) [inline horace_core/utilities/compute_bin_data.m:20]: this sentence does not make sense

**Review by cmarooney-stfc — CHANGES_REQUESTED** [review]

File: horace_core/sqw/PixelData/@PixelDataFileBacked/PixelDataFileBacked.m [hunk 2]
```diff
@@ -70,11 +70,19 @@
...
-        write_handle_ = []; % handle to the class, used to perform pixel
-        % copying in operations, involving all pixels
-        tmp_pix_obj_ = [];
+
+        % handle to the class used to perform pixel
+        write_handle_ = []; % copying in operations, involving all pixels
+
+        % handle-class holding tmp file produced by filebacked
+        % operations. If all referring classes go out of scope, the file
+        % gets deleted
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/sqw/PixelData/@PixelDataFileBacked/PixelDataFileBacked.m:84]: apparently, Matlab asignment for handle object increases ref counter so whatever surprizing it is, all works. 

I am unconsciously  using commas according to Russian rules, where sub-sentences are always separated by commas.

File: horace_core/sqw/PixelData/@PixelDataFileBacked/PixelDataFileBacked.m [hunk 3]
```diff
@@ -436,8 +446,10 @@ function saveobj(~)
...
     methods (Access = protected)
         function pix_data = get_raw_pix_data(obj,row_pix_idx,col_pix_idx)
+            % Overloaded part of get_raw_pix operation.
+            %
             % return unmodified pixel data according to the input indexes
-            % provided
+            % provided.
             % Inputs:
             % obj         -- initialized pixel data memory object
             % row_pix_idx -- indixes of pixels to return
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/sqw/PixelData/@PixelDataFileBacked/PixelDataFileBacked.m:455]: row and columt

File: horace_core/sqw/PixelData/@PixelDataMemory/PixelDataMemory.m [hunk 4]
```diff
@@ -230,8 +240,10 @@
...
     methods(Access=protected)
         function pix_data = get_raw_pix_data(obj,row_pix_idx,col_pix_idx)
+            % Overloaded part of get_raw_pix operation.
+            %
             % return unmodified pixel data according to the input indexes
-            % provided
+            % provided.
             % Inputs:
             % obj         -- initialized pixel data memory object
             % row_pix_idx -- indixes of pixels to return
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/sqw/PixelData/@PixelDataMemory/PixelDataMemory.m]: row and column, raw pixels -- mentioned above

File: horace_core/sqw/page_operations/PageOpBase.m [hunk 5]
```diff
@@ -1,58 +1,87 @@
...
+    % to the image and providing a unified interface for
+    % memory-/file-backed data.
+    %
+    % IMPORTANT:
+    % The operations can only be used by the algorithms which do not change
+    % the size and shape of the image.
+    % In addition, the operation must not change pixel coordinates in a
+    % way, which would violate the relation between image and the pixels
+    % i.e. alter which image bin a pixel would contribute 
+    % i.e. require rebinning or reordering of pixels.
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/sqw/page_operations/PageOpBase.m]: definitions.  Two equivalend definitions made in different words. Whatever you like better -- they say the same thing

File: horace_core/utilities/compute_bin_data.m (hunk 1 above)
Comment (abuts) [reply horace_core/utilities/compute_bin_data.m:20]: ? makes perfect sence to me.

**Review by cmarooney-stfc — CHANGES_REQUESTED** [review]

File: horace_core/sqw/PixelData/@PixelDataFileBacked/PixelDataFileBacked.m [hunk 6]
```diff
@@ -291,45 +318,15 @@ function saveobj(~)
...
-                obj.f_accessor_ = memmapfile(obj.full_filename, ...
-                    'format', obj.get_memmap_format(), ...
-                    'Repeat', 1, ...
-                    'Writable', true, ...
-                    'offset', obj.offset_);
-            end
+        function obj = finish_dump(obj,varargin)
+            % complete pixel write operation, close writing to the target
+            %  file and open pixel dataset for access operations.
+            obj = finish_dump_(obj);
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/sqw/PixelData/@PixelDataFileBacked/PixelDataFileBacked.m:324]: I do not see reason why we should not.

**Pull request merged**